packages/search-field/src/SearchField.js

```diff
@@ -38,17 +21,25 @@ class SearchField extends React.Component {
     this.props.debounceDelay
   );
 
-  onSearchChange = event => {
-    const { value } = event.target;
-    this.setState({
-      inputValue: value,
-      highlightedIndex: -1,
-      selectedItem: null
-    });
-    this.debouncedOnSearchChange(value);
-    if (!value) {
-      this.debouncedOnSearchChange.flush();
-      this.props.onChange();
+  setValueFromProps = () => {
+    const { value, items } = this.props;
+    const selectedItem = items && items.find(item => item.value === value);
+    if (selectedItem) {
+      this.setState({
+        inputValue: selectedItem.label,
```

Comment:
`inputValue` should be cleared when `value` prop changes to `null` or `undefined`.

I'm not sure what should be the behavior when it receives a `value` which is not within the `items`.  But it should not keep the last label.
Maybe we could dispatch an error, so we can track it with Sentry? What you think?

Reply:
I am not sure either if we should throw, would an exception break the app?

Reply:
I will just clear the input and the selection on both cases (update to no value, and update to non listed value)